### DISCUSSION TITLE
Unable to download from Chaika

### DISCUSSION THREAD

**Discussion by bossanovaorca** [Difegue/LANraragi#846] [Q&A]
When I add a Chaika URL to the downloader, I get the following error: `Error while processing archive.
(Address not available at /home/koyomi/lanraragi/script/../lib/LANraragi/Model/Upload.pm line 172. )`

is there a way around this issue?

**Comment by bossanovaorca** [comment]
can I get any help with this? even using the URL for the sample archive from the test data returns this error

after feeding the URLs into the downloader, the immediate response is `Error while processing archive.
(Address not available at /home/koyomi/lanraragi/script/../lib/LANraragi/Model/Upload.pm line 172. )`, even though the only thing printed in the logs is `Downloading...this will take some time`

**Reply by Difegue** [reply]
👋 Can your instance access the internet? You might want to test with some other plugins to make sure.

**Reply by bossanovaorca** [reply]
yes I am able to use the metadata plugins just fine (including chaika), as well as other downloaders. for some reason it's just chaika downloader that seems to have problems

**Reply by bossanovaorca** [reply]
is there anything I can try to debug this?

**Reply by bossanovaorca** [reply]
are you able to replicate the issue?

**Reply by Difegue** [reply]
Haven't looked into it yet, sorry 🙏 It'd be good to see if the chaika download URLs still work, eg `https://panda.chaika.moe/archive/(id)/download`. Maybe there's been a change in the website itself

**Reply by bossanovaorca** [reply]
so I did slightly more testing but unfortunately nothing gives me any more information than what we already have:

* `https://panda.chaika.moe/archive/(id)/` links work at random (fails more often than not with `Error while processing archive. (Unsupported File Extension (download))`)
* `https://panda.chaika.moe/archive/(id)/download` links don't work at all (`Error while processing archive. (Unsupported File Extension (download))`)
* for archives that I manually added and scraped metadata for, it works as expected (` Error while processing archive. (This file already exists in the Library. Enable replace duplicated archive in config to replace old ones.)`)

**Comment by Difegue** [comment]
Gave it a shot and everything seems to work as expected on my end -- I'm afraid I don't really have more troubleshooting steps, the `Address not available` error comes directly from the library I'm using to make network requests.

**Reply by bossanovaorca** [reply]
could this be an upstream issue where the library is having trouble parsing responses from Chaika?

**Comment by bossanovaorca** [comment]
I still often run into issues with `Unsupported File Extension` when downloading from various sites, including E-H/Ex-H. I don't believe they are truly file extension errors, since the archives are all `.rar` files so I suspect it may be some other issue, possibly related to a maximum number of jobs or something. Nothing in the application logs points to anything though, whether I try downloading in the application itself or via the Tsukihi extension

for some reason the title is being returned as `0start=1` on all E-H/Ex-H galleries I attempt this with, which then leads to the unsupported file extension error

**Reply by bossanovaorca** [reply]
to add on to this, the error still persists, it's just the line number has changed

```
 Error while processing archive.
(Inactivity timeout at /home/koyomi/lanraragi/script/../lib/LANraragi/Model/Upload.pm line 176. )
```

**Reply by Difegue** [reply]
I've seen this happening when you bundle a bunch of downloads at once in EH - Haven't had time to dig into it proper but it's on my list.

**Reply by bossanovaorca** [reply]
thank you, I appreciate the response. yes that is when I most commonly run into this issue as well. I'm glad that the issue has been replicated at least!
